iiasa/message_ix
# Use ‘genno’ for reporting

This PR adjusts `message_ix` to use the new package `genno` for reporting. See iiasa/ixmp#397 for a longer description/motivation.

The main change to `message_ix.reporting` is that features for `pyam` are removed. These have been moved upstream to `genno.compat.pyam`. The code is and tests are reduced, while retaining some tests to ensure that behaviour expected by `message_ix.reporting` still works.

## How to review

- Note that CI checks all pass.
- Try the code in combination with:
  - ~khaeru/genno#28~ merged; now genno >= 1.0.0
  - ~iiasa/ixmp#397~ merged; now ixmp `master`
  - iiasa/message_data#182
- View the revised documentation and ensure it is sufficient on a first read.

## PR checklist

- [x] Add temporary workflow config to install `genno` and `ixmp` from PR branches.
- [x] Add or expand tests.
- [x] Wait for required PRs to merge, then remove temporary workflow config.
  - [x] khaeru/genno#28
  - [x] iiasa/ixmp#397
